### PR TITLE
change-dependabot-interval-to-monthly

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Review by gregw — DISMISSED** [review]
Thanks

**Review by joakime — CHANGES_REQUESTED** [review]
This doesn't solve anything, in fact it just makes for more work, more time spent, and harsher demands on the CI system.

**Review by joakime — CHANGES_REQUESTED** [review]
Remove the schedule from 9.4.x, we should have it be scheduleless.

Set 10.0.x and 11.0.x to monthly.
See if you can trigger it on the first of each month. (see if the `"monthly"` literal is on the first of each month)

**Comment by joakime** [comment]
It would be keen if we could setup a schedule to be "the first Saturday of each month", that way the rush of CI builds will done over the weekend.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -75,7 +77,8 @@ updates:
     open-pull-requests-limit: 20
```
Comment (joakime) [inline .github/dependabot.yml]: Bump this up.
We have in 11.0.x an average of 4.2 dependabot PRs per week.
Lets make 11.0.x have a limit above 20, lets use 30.

Raw numbers from the start of the year.
```
11.0.x - Weekly updates
 
  is:pr created:>=2022-01-01 head:jetty-11.0.x
      2 open
     98 closed
   (4.4 PRs per week)
  is:pr created:>=2022-01-01 base:jetty-11.0.x label:dependencies 
      2 open
     93 closed
   (4.2 PRs per week)
  is:pr created:>=2022-01-01 base:jetty-11.0.x label:dependencies is:merged 
     56 merged
   (2.5 PRs per week)
  is:pr created:>=2022-01-01 base:jetty-11.0.x label:dependencies is:unmerged 
     39 un-merged
   (70% not merged)
```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -34,7 +35,8 @@ updates:
...
```
Comment (joakime) [inline .github/dependabot.yml]: Bump this up.
We have in 10.0.x an average of 6.8 dependabot PRs per week.
Lets make 10.0.x have a limit above 30, lets use 50.

Raw numbers from the start of the year.
```
10.0.x - Daily updates
   
  is:pr created:>=2022-01-01 head:jetty-10.0.x 
     18 open
    264 closed
    (12 PRs per week)
  is:pr created:>=2022-01-01 base:jetty-10.0.x label:dependencies 
      4 open
    151 closed
   (6.8 PRs per week)
  is:pr created:>=2022-01-01 base:jetty-10.0.x label:dependencies is:merged 
    127 merged
   (5.7 PRs per week) 
  is:pr created:>=2022-01-01 base:jetty-10.0.x label:dependencies is:unmerged 
     28 un-merged
   (22% not merged)
```

File: .github/dependabot.yml [hunk 3]
```diff
@@ -5,7 +5,8 @@ updates:
...
```
Comment (joakime) [inline .github/dependabot.yml]: Bump this up.
We have in 9.4.x an average of 5.8 dependabot PRs per week.
Lets make 9.4.x have a limit of 30.

Raw numbers from the start of the year.
```
9.4.x - Daily updates
  
  is:pr created:>=2022-01-01 base:jetty-9.4.x 
      4 open
    174 closed
   (7.9 PRs per week)
  is:pr created:>=2022-01-01 base:jetty-9.4.x label:dependencies
    128 closed
   (5.8 PRs per week)
  is:pr created:>=2022-01-01 base:jetty-9.4.x label:dependencies is:merged 
    105 merged
   (4.7 PRs per week)
  is:pr created:>=2022-01-01 base:jetty-9.4.x label:dependencies is:unmerged 
     23 un-merged
   (22% not merged)  
```

**Review by joakime — CHANGES_REQUESTED** [review]
We need to adjust the `open-pull-requests-limit` for the rush of PRs arriving monthly now.

**Comment by joakime** [comment]
Darn, looks like `schedule.interval` is a required field.

https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#scheduleinterval

Set them all to ...

``` yml
    schedule:
      interval: "monthly"
      day: "saturday"
      time: "02:00"
```

**Comment by joakime** [comment]
Our monthly runs will result in (on average) 67.2 PRs.  (calculation: `(5.8 + 6.8 + 4.2) * 4`)
Of which there's 2 (or 3) builds per PR.
So 134 to 201 builds.
Yeah, triggering these builds early on Saturday makes sense.

**Comment by olamy** [comment]
> Our monthly runs will result in (on average) 67.2 PRs. (calculation: `(5.8 + 6.8 + 4.2) * 4`) Of which there's 2 (or 3) builds per PR. So 134 to 201 builds. Yeah, triggering these builds early on Saturday makes sense.

`open-pull-requests-limit: 20`.
not sure how blocker is that. so that may delay a lot of updates :)

**Comment by joakime** [comment]
> `open-pull-requests-limit: 20`.
> not sure how blocker is that. so that may delay a lot of updates :)

Check the comments on the PR, I posted some suggestions on new values for these.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -31,10 +33,12 @@ updates:
 
   - package-ecosystem: "maven"
     directory: "/"
-    open-pull-requests-limit: 20
+    open-pull-requests-limit: 50
```
Comment (gregw) [inline .github/dependabot.yml:36]: Is there a reason this is 50 and the others 30?   Expectations of more changes here?

**Review by gregw — APPROVED** [review]

File: .github/dependabot.yml (hunk 4 above)
Comment (joakime) [reply .github/dependabot.yml:36]: This is covered in [a prior comment](https://github.com/eclipse/jetty.project/pull/8140#discussion_r892882429).

**Comment by olamy** [comment]
@joakime I need your approval as you requested changes. thanks

**Review by joakime — APPROVED** [review]

**Comment by olamy** [comment]
all good as the PR has even proper labels 🤣 😜.  
I will merge it.

**Pull request merged**